### PR TITLE
{Security} Update security module to track 2 sdk

### PR DESCRIPTION
**Pull request by a7mad-3bbas**
**Description**
 This PR is to update the security module to track 2 sdk 

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
security

File: src/azure-cli/azure/cli/command_modules/security/custom.py [hunk 1]
```diff
@@ -23,7 +28,7 @@
 def list_security_tasks(client, resource_group_name=None):
 
     for loc in client.locations.list():
-        client.config.asc_location = loc.name
+        client._config.asc_location = loc.name
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/security/custom.py:31]:
```suggestion
        client._config.asc_location = loc.name  # pylint: disable=protected-access
```

File: src/azure-cli/azure/cli/command_modules/security/custom.py [hunk 2]
```diff
@@ -547,7 +552,7 @@ def list_security_sub_assessments(client, assessment_name=None, assessed_resourc
 def get_security_sub_assessment(client, resource_name, assessment_name, assessed_resource_id=None):
 
     if assessed_resource_id is None:
-        assessed_resource_id = '/subscriptions/' + client.config.subscription_id
+        assessed_resource_id = '/subscriptions/' + client._config.subscription_id  # pylint: disable=protected-access
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/security/custom.py:555]: same issue as above

File: src/azure-cli/azure/cli/command_modules/security/custom.py [hunk 3]
```diff
@@ -431,9 +436,9 @@ def _get_va_sql_resource_id(vm_resource_id, server_name, database_name, vm_name,
     raise MutuallyExclusiveArgumentError('Please specify all of (--vm-name, --agent-id, --vm-uuid) for On-Premise resources, or none, other resource types')
 
 
-def _get_va_sql_api_version():
+# def _get_va_sql_api_version():
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/security/custom.py:439]: Remove this function if no use

**Review by jsntcy — APPROVED** [review]

**Pull request merged**